packages/ai/brain_config.py

```
@@ -1215,3 +1216,85 @@ async def refresh_brain_config_cache() -> BrainConfig:
+        return settings.is_north_mini_code_default
+    except Exception:  # noqa: BLE001 — defensive; flag defaults ON
+        return True
+
+
+def _active_primary_provider() -> str | None:
+    """Best-effort read of the active brain's primary provider (or ``None``)."""
+    try:
+        if _store is not None and _store._cache is not None:
+            return str(getattr(_store._cache, "primary_provider", "")).strip().lower() or None
+    except Exception:  # noqa: BLE001 — defensive
+        pass
```

Ruff (0.15.21)

[error] `try`-`except`-`pass` detected, consider logging the exception

(S110)